db/crud.go

```diff
@@ -896,11 +896,14 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 }
 
 // updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering. mouMatch represents if the _mou.cas == doc.cas
-func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent DocUpdateType, mouMatch bool) (*Document, error) {
+func (db *DatabaseCollectionWithUser) updateHLV(ctx context.Context, d *Document, docUpdateEvent DocUpdateType, mouMatch bool) (*Document, error) {
 
 	hasHLV := d.HLV != nil
 	if d.HLV == nil {
 		d.HLV = &HybridLogicalVector{}
+		base.DebugfCtx(ctx, base.KeySGTest, "No existing HLV for doc %s", base.UD(d.ID))
```

Comment:
It feels like we should be putting these under the CRUD log key, rather than something test-specific?

```suggestion
		base.DebugfCtx(ctx, base.KeyCRUD, "No existing HLV for doc %s", base.UD(d.ID))
```

Reply:
I was very reluctant to put these under the existing CRUD key - these are going to be _very_ noisy for anything but topology tests and we wouldn't have this on in production. I am very open to ideas of how to solve the problem.

I thought about:

- Putting these at `KeyCRUD` but Trace, but then if you want this logging, you have to opt-in for trace logging for everything, and this is prohibitive to analyze for.
- Create a specific topology test key (but KeySGTest is so ill used anyway, I thought it was the same)
- Create a function like `base.TestInfofCtx` that would compile away if `cb_sg_devmode` was not used - we pretty much don't need this outside of running a specific test environment, that I'm aware of.

That said, if something _did_ go wrong in production or say a QE/perf test this is useful to know about.

Reply:
I'm certainly open to keeping at debug with a version-vector specific log key, at least for the beta and early 4.x versions until we're more confident about the implementation.

This is more reason to drop the SGTest key, I think we're going to need it for QE and beta issues.